usaspending_api/common/serializers.py

```diff
@@ -7,83 +7,127 @@ class LimitableSerializer(serializers.ModelSerializer):
 
     """Extends the model serializer to support field limiting."""
     def __init__(self, *args, **kwargs):
-        # next two lines are deprecated and will be removed
-        # once all views inherit from a generic class or are
-        # refactored into viewsets (i.e., once the serializer
-        # consistently has direct access to the request)
-        include_fields = kwargs.pop('fields', None)
-        exclude_fields = kwargs.pop('exclude', None)
+        # Grab any kwargs include and exclude fields, these are typically
+        # passed in by a parent serializer to the child serializer
+        kwargs_has_include = 'fields' in kwargs
```

Comment:
This allows us to set the fields by passing them in via context if we want to. If they are not present in the context, we check the request object. Typically, the process flows:

Top level serializer checks kwargs, finds nothing
Top level serializer checks request, finds fields + excludes
Top level serializer passes down relevant child fields via their context
Then the process repeats all the way down the nesting chain.